basic-javascript/project-1.spec.js

```diff
@@ -1,3 +1,332 @@
+  expect(helpers.cube([])).toBe(false);
+  expect(helpers.cube({})).toBe(false);
+  expect(helpers.cube(NaN)).toBe(false);
+});
+
+//raiseToPower
+
+it("should return the square of a number", () => {
+  expect(helpers.raiseToPower(2, 3)).toBe(8);
+});
+
+it("should convert strings to numbers if possible", () => {
+  expect(helpers.raiseToPower("2c", 3)).toBe(8);
+});
+
+it("should return false if passed an argument other than a string or number", () => {
+  expect(helpers.raiseToPower([], 2)).toBe(false);
+  expect(helpers.raiseToPower({}, 2)).toBe(false);
+  expect(helpers.raiseToPower(NaN, 2)).toBe(false);
+});
+
+//roundNumber
+
+it("should return a number rounded to the nearest integer", () => {
+  expect(helpers.roundNumber(2.7)).toBe(3);
+});
+
+it("should convert strings to numbers if possible", () => {
+  expect(helpers.roundNumber("2.555c")).toBe(3);
+});
+
+it("should return false if passed an argument other than a string or number", () => {
+  expect(helpers.roundNumber([])).toBe(false);
+  expect(helpers.roundNumber({})).toBe(false);
+  expect(helpers.roundNumber(NaN)).toBe(false);
+});
+
+//addExclamationPoint
+
+it("return a string with an an exclamation point added to the end", () => {
+  expect(helpers.addExclamationPoint("Here's Johnny")).toBe("Here's Johnny!");
+});
+
+it("should convert its argument to a string if passed a number or array", () => {
+  expect(helpers.addExclamationPoint(222)).toBe("222!");
+  expect(helpers.addExclamationPoint(["dog"])).toBe("dog!");
+});
+
+//combineNames
+
+it("should combine two name strings into one", () => {
+  expect(helpers.combineNames("cat", "dog")).toBe("cat dog");
+});
+
+it("should return false if either one or both arguments are not a string", () => {
+  expect(helpers.combineNames(undefined, "cat")).toBe(false);
+  expect(helpers.combineNames(null, 33)).toBe(false);
+  expect(helpers.combineNames("dog", null)).toBe(false);
+  expect(helpers.combineNames(33, undefined)).toBe(false);
+});
+
+//getGreeting
+
+it("should return a greeting strings into one", () => {
+  expect(helpers.getGreeting("nurse")).toBe("Hello nurse!");
+});
+
+it("should return false if its argument is not a string", () => {
+  expect(helpers.getGreeting(undefined)).toBe(false);
+  expect(helpers.getGreeting(33)).toBe(false);
+  expect(helpers.getGreeting({})).toBe(false);
+});
+
+//getRectangleArea
+
+it("return the area of a rectangle as a number", () => {
+  expect(helpers.getRectangleArea(5, 2)).toBe(10);
+});
+
+it("should convert strings to numbers if possible", () => {
+  expect(helpers.getRectangleArea("5vvv", 2)).toBe(10);
+  expect(helpers.getRectangleArea(5, "2sss")).toBe(10);
+});
+
+//getTriangleArea
+
+it("return the area of a triangle as a number", () => {
+  expect(helpers.getTriangleArea(5, 2)).toBe(5);
+  expect(helpers.getTriangleArea(3.55, 2.55)).toBeCloseTo(4.53);
+  expect(helpers.getTriangleArea(-2, 2)).toBe(-2);
```

Comment:
Good job using toBeCloseTo here and with getCircleArea, which involves pi. This way, you don't have to worry about how many decimal places javascript is going to calculate.